lib/pacemaker/pcmk_sched_constraints.c

```diff
@@ -2279,8 +2281,38 @@ unpack_rsc_order(xmlNode * xml_obj, pe_working_set_t * data_set)
     return TRUE;
 }
 
+/*!
+ * \internal
+ * \brief Return the boolean influence corresponding to configuration
+ *
+ * \param[in] coloc_id     Colocation XML ID (for error logging)
+ * \param[in] rsc          Resource involved in constraint (for default)
+ * \param[in] influence_s  String value of influence option
+ *
+ * \return true if string evaluates true, false if string evaluates false,
+ *         or value of resource's critical option if string is NULL or invalid
+ */
+static bool
+unpack_influence(const char *coloc_id, const pe_resource_t *rsc,
+                 const char *influence_s)
+{
+    if (influence_s != NULL) {
+        int influence_i = 0;
+
+        if (crm_str_to_boolean(influence_s, &influence_i) < 0) {
```

Comment:
I might be missing some subtle distinction here, but couldn't you use `if (!crm_is_true(influence_s))` and skip the intermediate variable?

Reply:
Necessary to use pe_rsc_critical as the default